KubeArmor/core/hook_handlier_non_k8s.go

```diff
@@ -0,0 +1,268 @@
+		dm.Logger.Warnf("Failed to cleanup ka.sock: %v", err)
+	}
+
+	socket, err := net.Listen("unix", listenPath)
+	if err != nil {
+		dm.Logger.Warnf("Failed listening on ka.sock: %v", err)
+		return
+	}
+
+	// Set the permissions of ka.sock to 777 so that rootless podman with user level priviledges can also communicate with the socket
+	if err := os.Chmod(listenPath, 0777); err != nil {
+		dm.Logger.Warnf("Failed to set permissions on %s: %v", listenPath, err)
+	}
+
+	defer socket.Close()
+	defer os.Remove(listenPath)
+	ready := &atomic.Bool{}
+
+	for {
+		conn, err := socket.Accept()
+		if err != nil {
+			dm.Logger.Warnf("Error accepting socket connection: %v", err)
+		}
+
+		go dm.handleNonK8sConn(conn, ready)
+	}
+
+}
+
+// handleNonK8sConn gets container details from container hooks.
+func (dm *KubeArmorDaemon) handleNonK8sConn(conn net.Conn, ready *atomic.Bool) {
+	// We need to makes sure that no containers accepted until all containers created before KubeArmor
+	// are sent first. This is done mainly to avoid race conditions between hooks sending in
+	// data that some containers were deleted only for process responsible for sending previous containers
+	// to send that these containers are created. Which will leave KubeArmor in an incorrect state.
+	defer conn.Close()
+	buf := make([]byte, 4096)
+
+	for {
+		n, err := conn.Read(buf)
+		if err == io.EOF {
+			return
+		} else if err != nil {
+			dm.Logger.Warnf("Error reading connection: %v", err)
+		}
+
+		data := tp.HookRequest{}
+
+		err = json.Unmarshal(buf[:n], &data)
+		if err != nil {
+			dm.Logger.Warnf("Error unmarshalling: %v", err)
+		}
+
+		if data.Detached {
+			// we want KubeArmor to start accepting containers after
+			// all previous container are set
+			defer ready.Store(true)
+		} else if !ready.Load() {
+			_, err = conn.Write([]byte("err"))
+			if err == io.EOF {
+				return
+			} else if err != nil {
+				log.Println(err)
+				return
+			}
+			continue
+		}
+		_, err = conn.Write([]byte("ok"))
+		if err == io.EOF {
+			return
+		} else if err != nil {
+			log.Println(err)
+			return
+		}
+
+        // Handle the container create or delete event
+        if data.Operation == tp.HookContainerCreate {
+            dm.UpdateContainer(data.Container.ContainerID, data.Container, "create")
+        } else {
+            dm.UpdateContainer(data.Container.ContainerID, data.Container, "destroy")
+        }
+
+	}
+}
+
+// UpdateContainer Function
+func (dm *KubeArmorDaemon) UpdateContainer(containerID string, container tp.Container, action string) bool {
+
+	if action == "create" {
+
+		if container.ContainerID == "" {
+			return false
+		}
+
+		endPoint := tp.EndPoint{}
+
+		dm.ContainersLock.Lock()
+		if _, ok := dm.Containers[container.ContainerID]; !ok {
+			dm.Containers[container.ContainerID] = container
+			dm.ContainersLock.Unlock()
+
+			containerLabels, containerIdentities := common.GetLabelsFromString(container.Labels)
+			dm.EndPointsLock.Lock()
```

Comment:
while there's no performance impact however if we move this lock to the L#168 where we're updating the endpoint. wdyt?